src/mlio/record_readers/csv_record_reader.cxx

```diff
@@ -93,6 +99,18 @@ csv_record_reader::read_line(memory_slice &chunk, bool ignore_leftover)
     for (auto pos = chrs.begin(); pos != chrs.end(); ++pos) {
         auto chr = *pos;
 
+        // Check that we are within the size limit for a row.
+        auto bytes_read = sizeof(char) * as_size(pos - chrs.begin());
+        if (params_->max_line_length &&
+            bytes_read > *params_->max_line_length) {
+            throw record_too_large_error{fmt::format(
+                "Line exceeds the maximum line length of {1:n}. "
+                "Line begins: '{0}'",
+                std::string(chrs.data(),
```

Comment:
I see that you have started using `std::string_view` in `text_line.cxx`, but here you still construct an `std::string`.